api/swagger/swagger.yaml

```diff
@@ -0,0 +1,429 @@
+      consumes:
+        - application/json
+      produces:
+        - application/json
+      parameters:
+      - name: id
+        in: path
+        description: Post id
+        type: string
+        format: uuid
+        required: true
+      - name: body
+        in: body
+        required: true
+        schema:
+          $ref: '#/definitions/Post'
+      responses:
+        200:
+          $ref: "#/responses/ok"  
+        400:
+          $ref: '#/responses/badRequest'
+        403:
+          $ref: '#/responses/forbidden'
+    delete:
+      tags:
+        - posts
+      summary: Delete post from the store
+      consumes:
+        - application/json
+      produces:
+        - application/json
+      parameters:
+      - name: id
+        in: path
+        description: Post id
+        type: string
+        format: uuid
+        required: true
+      responses:
+        200:
+          $ref: "#/responses/ok"  
+        400:
+          $ref: '#/responses/badRequest'
+        403:
+          $ref: '#/responses/forbidden'
+  /posts/{id}/comments:        
+    get:
+      tags:
+        - posts
+      summary: Get post comments
+      produces:
+        - application/json
+      parameters:
+      - name: id
+        in: path
+        description: Post id
+        type: string
+        format: uuid
+        required: true
+      responses:
+        200:
+          schema:
+            type: array
+            items:
+              $ref: '#/definitions/Comments'
+          description: ''
+        400:
+          $ref: '#/responses/badRequest'
+        403:
+          $ref: '#/responses/forbidden'
+        404:
+          $ref: '#/responses/notFound'
+    post:
+      tags:
+        - posts
+      summary: Add comment to post
+      consumes:
+        - application/json
+      produces:
+        - application/json
+      parameters:
+      - name: id
+        in: path
+        description: Post id
+        type: string
+        format: uuid
+        required: true
+      - name: body
+        in: body
+        required: true
+        schema:
+          $ref: '#/definitions/Comment'
+      responses:
+        200:
+          $ref: "#/responses/ok"  
+        400:
+          $ref: '#/responses/badRequest'
+        403:
+          $ref: '#/responses/forbidden'
+    delete:
+      tags:
+        - posts
+      summary: Delete post comments
+      consumes:
+        - application/json
+      produces:
+        - application/json
+      parameters:
+      - name: id
+        in: path
+        description: Post id
+        type: string
+        format: uuid
+        required: true
+      responses:
+        200:
+          $ref: "#/responses/ok"  
+        400:
+          $ref: '#/responses/badRequest'
+        403:
+          $ref: '#/responses/forbidden'
+  /user/{username}:
+    get:
+      tags:
+        - user
+      summary: Get user by user name
+      produces:
+        - application/json
+      parameters:
+      - name: username
```

Comment:
why not id here?

Reply:
It's an open question :) I'm choosing between username and id. But probably id would be more convenient.